src/main/java/lotto/domain/LottoTicketMachine.java

```diff
@@ -0,0 +1,31 @@
+package lotto.domain;
+
+import java.util.List;
+import lotto.ticketingway.AutoWay;
+import lotto.ticketingway.LottoTicketingWay;
+import lotto.ticketingway.ManualWay;
+
+public class LottoTicketMachine {
+
+    private final static LottoTicketMachine instance = new LottoTicketMachine();
+
+    private LottoTicketMachine() {
+    }
+
+    public static LottoTicketMachine getInstance() {
+        return instance;
+    }
+
+    private List<LottoTicket> issueTickets(IssueInput input, LottoTicketingWay way) {
+        return way.issueLottoTickets(input);
+    }
+
+    public AutoLottoTickets issueTicketsByAutoWay(IssueInput input) {
```

Comment:
```suggestion
    public AutoLottoTickets buyTicketsByAutoWay(IssueInput input) {
```
와 같은 네이밍이 좀 더 직관적이지 않을까요?

Reply:
"로또티켓기계(LottoTicketMachine)" 는 로또를 "발행해주는" 일을 하지, "로또는 사는" 행위는 하지 않는다고 생각해서 이러한 이름으로 짓게되었습니다. 돈에 관련된 파라메터도 없구요.
실제로 돈을 받는쪽은 "로또게임업자(LottoGameVendor)" 이고, 이클래스에는 "buyLottos"라는 메서드가 있습니다.
저는 이 이름이 그렇게 나쁘지 않다고 생각합니다만 어떻게 생각하시나요?

Reply:
그런 맥락이라면 괜찮을 것 같습니다!

Reply:
답변 감사합니다!